### PR TITLE
added RenameColumnsTransformer to misc module

### PR DESCRIPTION
**Pull request by liamholmes31**
set up simple transformer for renaming (or copying columns) https://github.com/azukds/tubular/issues/633

a bit niche, but have seen cases where it is/could be used so worth just bringing in I think

### CODE REVIEW

File: tests/pipeline/test_pipeline_dump_and_load_json.py [hunk 1]
```diff
@@ -13,8 +13,7 @@ class FakeTransformer:
 class TestPipelineDumpAndLoadJson:
     """Tests for dump_pipeline_to_json() and load_pipeline_from_json()."""
 
-    @staticmethod
-    def test_dump_pipeline_then_load_pipeline():
+    def test_dump_pipeline_then_load_pipeline(self):
```
Comment (liamholmes31) [inline tests/pipeline/test_pipeline_dump_and_load_json.py:16]: just cleaning up the leftover staticmethods spotted here, a previous PR removed them from the rest of the test suite

File: tubular/misc.py [hunk 2]
```diff
@@ -165,6 +166,235 @@ def transform(self, X: DataFrame) -> DataFrame:
...
+
+        Raises
+        ------
+            ValueError: if provided columns are not keys of
+                provided new_column_names
+
+        """
+        super().__init__(columns=columns, **kwargs)
+
+        msg = f"{self.classname()}: all provided columns must appear as keys in new_column_names"  # noqa: E501
```
Comment (hariinduu) [inline tubular/misc.py:271]: Q: What could happen if you provide an existing column name as a new_column name?

File: tests/misc/test_RenameColumnsTransformer.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+class TestFit(GenericFitTests):
+    """Generic tests for RenameColumnsTransformer.fit()"""
+
+    @classmethod
+    def setup_class(cls):
+        cls.transformer_name = "RenameColumnsTransformer"
+
+
+class TestTransform(GenericTransformTests):
```
Comment (hariinduu) [inline tests/misc/test_RenameColumnsTransformer.py:43]: S: Should we add a test with an empty data frame to see whether it still renaming the column?

File: tests/misc/test_RenameColumnsTransformer.py
Comment (hariinduu) [inline tests/misc/test_RenameColumnsTransformer.py:1]: Q: Also thinking whether we should test special scenarios such as  case sensitivity  (For example a -> A ) , special characters in new_column names, python keywords in new_column names and overlapping it with indexes

File: tubular/misc.py (hunk 2 above)
Comment (liamholmes31) [reply tubular/misc.py:271]: good question, added handling here https://github.com/azukds/tubular/pull/647/changes/ff1cc05e7f8c31a3b72f2562fba9e698b058948c

File: tests/misc/test_RenameColumnsTransformer.py
Comment (liamholmes31) [reply tests/misc/test_RenameColumnsTransformer.py:1]: updated the test case here https://github.com/azukds/tubular/pull/647/changes/ff1cc05e7f8c31a3b72f2562fba9e698b058948c

File: tests/misc/test_RenameColumnsTransformer.py (hunk 3 above)
Comment (liamholmes31) [reply tests/misc/test_RenameColumnsTransformer.py:43]: added here https://github.com/azukds/tubular/pull/647/changes/853adadce20aa94c49eb74887d6657d1abbb36db
